### ISSUE TITLE
jsxDevRuntimeExports.jsxDEV is not a function

### ISSUE THREAD

**Issue by samesfahani-tuplehealth** [vitejs/vite-plugin-react-swc#154]
I'm in the process of migrating our codebase from CRA to Vite and am getting close, but I am hitting an error now only when I `vite build && vite preview` for production. The dev server seemingly loads everything fine.

```
index-85d3be4e.js:292558 Uncaught TypeError: jsxDevRuntimeExports.jsxDEV is not a function
    at index-85d3be4e.js:292558:121
```
As I understand it, the jsx runtime should be `automatic`, and as I said, it works in dev, so what could be the difference when building for production?

**Comment by ArnaudBarre** [comment]
`jsxDev` should not be present in the build output, but I can't help you without a repro

**Comment by samesfahani-tuplehealth** [comment]
Your comment did somewhat help. I don't know why (still new to Vite so perhaps intended), but I need to specify `NODE_ENV=production` in the build script (even though the `mode` is production?). Either way, trivial issue (and not related to this plugin, probably).

**Comment by ArnaudBarre** [comment]
Yeah the `mode` in Vite is different from `NODE_ENV`, it something that you can customize to have staging mode or things like that.
I think `NODE_ENV` is set to  `production` in vite build, and it works without configuration on the default template so you may have something that override this somewhere in your build process and you just over-override, which is fine but if you find the root cause you can probably simplify things

**Comment by samesfahani-tuplehealth** [comment]
Had `NODE_ENV=development` in a shell rc file in case it helps anyone. Thanks for your help.
